sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala

```diff
@@ -1106,6 +1106,41 @@ class StreamingQuerySuite extends StreamTest with BeforeAndAfter with Logging wi
     }
   }
 
+  test("union in streaming query of append mode without watermark") {
+    val inputData1 = MemoryStream[Int]
+    val inputData2 = MemoryStream[Int]
+    withTempDir { dir =>
+      withTempView("s1", "s2") {
+        inputData1.toDF().createOrReplaceTempView("s1")
+        inputData2.toDF().createOrReplaceTempView("s2")
+        val unioned = spark.sql(
+          "select s1.value from s1 union select s2.value from s2")
+        checkExceptionMessage(unioned, dir)
+      }
+    }
+  }
+
+  test("distinct in streaming query of append mode without watermark") {
+    val inputData = MemoryStream[Int]
+    withTempDir { dir =>
+      withTempView("deduptest") {
+        inputData.toDF().toDF("value").createOrReplaceTempView("deduptest")
+        val distinct = spark.sql("select distinct value from deduptest")
+        checkExceptionMessage(distinct, dir)
+      }
+    }
+  }
+
+  private def checkExceptionMessage(df: DataFrame, dir: File): Unit = {
+    val exception = intercept[AnalysisException](
+      df.writeStream
+        .option("checkpointLocation", dir.getCanonicalPath)
+        .start(dir.getCanonicalPath))
```

Comment:
Even we know the query will throw error, it'd be ideal to follow the normal path, ensuring we have different dir for output and checkpoint.

Reply:
Make sense, done in ca16b67.